src/utils/rlp.cairo

```diff
@@ -60,7 +62,11 @@ namespace RLP {
 
         local len_bytes_count = prefix - 0xf7;
         let list_len = Helpers.bytes_to_felt(len_bytes_count, data + 1);
-        return (TYPE_LIST, 1 + len_bytes_count, list_len);
+        tempvar offset = 1 + len_bytes_count;
+        assert [range_check_ptr] = offset;
+        assert [range_check_ptr + 1] = list_len;
+        let range_check_ptr = range_check_ptr + 2;
+        return (TYPE_LIST, offset, list_len);
```

Comment:
To be sure to understand: `len_bytes_count` can be arbitrary hence `Helpers.bytes_to_felt(len_bytes_count, data + 1);` can overflow silently, hence `list_len` can pass the `range_check`.
But the check in `decode_raw` on the `data_len` allows to be sure the `list_len` is correct. 
If this is the case, `decode_type` should be marked as unsafe (`decode_type_unsafe`) and add a comment that the length of the data needs to be checked.

Reply:
I wonder if bytes_to_felt should be marked as unsafe also